wrangler/helpers/labware_helpers.py

```
@@ -55,22 +63,37 @@ def wrangle_labware(labware_barcode: str) -> Tuple[Dict[str, str], int]:
         labware_type = determine_labware_type(labware_barcode, results)
         logger.info(f"Determined labware type: {labware_type}")
 
-        if labware_type == TUBE_RACK:
+        # Assuming study is the same for all wells/tubes in a container
+        study_name = results[0]["study"]
+        logger.info(f"Study name: {study_name}")
```

Oh so does this pull request also tick off GPL-441 (https://github.com/sanger/wrangler/issues/41)?

Just wondering - would it be worth keeping a default study in case this column is blank? In case of processing old data where it's not populated or something?

---

I guess it does tick off GPL-441 as well.

The `study` column in the `cgap_heron` table is non-null, so it should never be blank, including for old data. I think I'd rather the wrangler failed if the MLWH data wasn't as expected, rather than use a default instead, but am open to suggestions.

---

OK no probs then 👍